arcgis-ios-sdk-samples/Edit data/Edit feature attachments/EditFeatureAttachmentsViewController.swift

```diff
@@ -0,0 +1,121 @@
+// Copyright 2016 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class EditFeatureAttachmentsViewController: UIViewController {
+    
+    @IBOutlet private weak var mapView: AGSMapView!
+    
+    private var featureLayer: AGSFeatureLayer?
+    private var lastQuery: AGSCancelable?
+    
+    private var selectedFeature: AGSArcGISFeature?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = [
+            "EditFeatureAttachmentsViewController",
+            "AttachmentsTableViewController"
+        ]
+        
+        let map = AGSMap(basemap: .oceans())
+        //set initial viewpoint
+        map.initialViewpoint = AGSViewpoint(
+            center: AGSPoint(x: 0, y: 0, spatialReference: .webMercator()),
+            scale: 100000000
+        )
+        
+        let featureServiceURL = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer/0")!
+        let featureTable = AGSServiceFeatureTable(url: featureServiceURL)
+        let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+        self.featureLayer = featureLayer
+        
+        map.operationalLayers.add(featureLayer)
+        
+        mapView.map = map
+        mapView.touchDelegate = self
+    }
+
+    // MARK: - Navigation
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let navController = segue.destination as? UINavigationController,
+            let controller = navController.viewControllers.first as? AttachmentsTableViewController {
+            controller.feature = selectedFeature
+        }
+    }
+}
+
+extension EditFeatureAttachmentsViewController: AGSGeoViewTouchDelegate {
+    
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        
+        if let lastQuery = lastQuery {
+            lastQuery.cancel()
+        }
+        
+        //hide the callout
+        mapView.callout.dismiss()
+        
+        guard let featureLayer = featureLayer else {
```

Comment:
This seems like such a silly check to do every time because we know there will always be a feature layer. It would be nice to either make `featureLayer` an implicitly unwrapped optional or make it non-optional by creating it when the view controller is initialized.